File: demo_leak/demo_leak/ViewController.m
//
//  ViewController.m
//  demo_leak
//
//  Created by Lwgfangz on 14-7-23.
//  Copyright (c) 2014年 aiDriving.com. All rights reserved.
//

#import "ViewController.h"
#import "largerObjct.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)action:(id)sender
{
    [[largerObjct alloc]init];
}
- (IBAction)actionFree:(id)sender
{
    [[[largerObjct alloc]init] autorelease] ;
    //[[[UIImage alloc]init] autorelease] ;
    //[[UIImage alloc]init] ;
}

@end

File: demo_swift11/demo_swift11/ViewController.swift
//
//  ViewController.swift
//  demo_swift11
//
//  Created by Lwgfangz on 14-9-19.
//  Copyright (c) 2014年 Lwgfangz. All rights reserved.
//

import UIKit
// Playground - noun: a place where people can play

import UIKit

var str = "Hello, playground"


class My
{
    var a:Int;
    
    init(b:Int)
    {
        self.a = 1;
        println(b);
    }
    func cry()
    {
        var b:Int=9;
        b = 9;
        NSLog("asfd");
        var c = b;
    }
}



class ViewController: UIViewController
{
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        
        var  a = My(b:3);
        a.cry();
        
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

